test/mongodb.test.js

```diff
@@ -547,6 +687,200 @@ describe('mongodb connector', function() {
     done();
   });
 
+  context('nested decimal props', function() {
+    it('should create/update instance for array of decimal props', function(done) {
+      modelWithDecimalArray.create({
+        'randomReview': [
+          '3.5',
+          '4.5',
+          '4.0',
+        ],
+      }, function(err, inst) {
+        inst.toObject().randomReview[0].should.equal(Decimal128.fromString('3.5'));
```

Comment:
I am little bit confused about this assertion. Are you sure that `equal` check is verifying whether both values are instances of `Decimal128`? Shouldn't you use `instanceof` operator instead?

I guess the more important question is whether you have seen these new tests fail on this assertion and what were the error messages reported?

Reply:
Yes the tests fail like so:
```
  1) mongodb connector
       nested decimal props
         should create/update instance for array of decimal props:
     Uncaught AssertionError: expected '3.5' to be Decimal128 {
  _bsontype: 'Decimal128',
  bytes: Buffer [ 23, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 3e, 30 ]
}
```
I think using `instanceOf` operator is better, but like you mentioned above, I might have to use a custom `execute` call to grab the raw data from mongodb.

Reply:
> ```
>  Uncaught AssertionError: expected '3.5' to be Decimal128 {
>  _bsontype: 'Decimal128',
>  bytes: Buffer [ 23, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 3e, 30 ]
> }
> ```

Nice!